documentation/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.md

```diff
@@ -0,0 +1,263 @@
+
+Each `VMediaTag` should be a `VButton` with `as="VLink"` and should link to the
+localized page for the tag collection. Figma link:
+https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?type=design&node-id=1200-56515&mode=design&t=nCX20BtJYqMOFAQm-4
+[VMediaTag component](https://github.com/WordPress/openverse/blob/c7b76139d5a001ce43bde27805be5394e5732d1a/frontend/src/components/VMediaTag/VMediaTag.vue)
+`<VButton variant="filled-gray" size="small" class="label-bold" as="VLink" href="href">{{tag}}</VButton>`.
+It should accept `href` as a prop.
+
+### Other interface changes
+
+#### Update the "by" area of the single result page
+
+The "by creator" line should be replaced with a section that is
+horizontally-scrollable on mobile, and contains the `VCreatorLink` and
+`VSourceLink` components.
+
+#### Update links in the "information" section
+
+Add external link icon to the provider link; add creator link with external link
+icon The links to creator in the image and audio single result pages should have
+an "external link" icon.
+[image creator link](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/pages/image/_id/index.vue#L62-L73)
+and
+[audio creator link](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/components/VAudioTrack/layouts/VFullLayout.vue#L32-L41).
+Audio creator link should also be updated to match the image creator link. It
+should be a conditional component: `VLink` if the `creator_url` is truthy and
+`span` if the `creator_url` is falsy.
+
+Currently, the `foreign_landing_url` is linked to the "source" in the image page
+and "provider" in the audio page. The audio page should be updated to match the
+image page: the `foreign_landing_url` link should be added to the "source", not
+provider.
+[Audio provider link](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/components/VAudioDetails/VAudioDetails.vue#L61C1-L63),
+[image source link](https://github.com/WordPress/openverse/blob/35f08e5710d59f6c10f0cf54103fcce151adfefe/frontend/src/components/VImageDetails/VImageDetails.vue#L26-L28)
+
+### Store changes
+
+#### Update the `searchBy` filter in the `search` store
+
+Currently, this filter is shaped as other filters: it is a list of objects with
+`code`, `name` and `checked` properties. This is a possible value now: `{
+searchBy: { code: "creator", name: "filters.searchBy.creator", checked: true }}.
+
+Since this filter is mutually exclusive (you cannot search by both the creator
+and the tag, for example), this shape is very difficult to update.
+
+It is better to set the `searchBy` filter to have one of the possible values:
+`{ searchBy: <null|creator|source|tag> }`.
+
+We should also create a new method, `setSearchBy`, in the `search` store, that
+would allow directly set the `searchBy` filter value.
```

Comment:
Can you clarify whether this is just a DevEx/code-quality refactor or if it's necessary for the new features? I don't fully understand why this particular change is relevant to the creator/tag/source pages. Are they going to use the same search store and filters? If so, that would be helpful background to have somewhere as well (may have missed it though?).

Reply:
Yes, I actually re-used the same search store in the draft implementation that I linked in the issue description. But to get the store to handle different `searchBy` values, we should add this change.
Thank you for asking this question! I realized that I didn't write anything about the data side: what API calls will be made, how the stores will be used and how they need to be updated. I'll add this during the revision round.